File: Examples/Blueprint Example (Obj-C)/Blueprint Example (Obj-C)/Owner.m
//
//  Owner.m
//  Blueprint Example (Obj-C)
//
//  Created by Hunter on 5/12/16.
//  Copyright © 2016 Blueprint Project. All rights reserved.
//

#import "Owner.h"
#import "Pet.h"

@implementation Owner

-(BPSingleRecordPromise *)findPetWithName:(NSString *)name
{
    NSDictionary *query = @{@"name": name};
    return [Pet findOne:query];
}

@end

File: Examples/Blueprint Example/Models/Owner.swift
//
//  Owner.swift
//  Blueprint Example
//
//  Created by Hunter on 5/1/16.
//  Copyright © 2016 Blueprint Project. All rights reserved.
//

import Cocoa

class Owner: BPModel {

    func findPetWithName(_ name: String!) -> BPSingleRecordPromise {
        let query = ["name": name as NSObject, "$limit": 1 as NSObject] as [String: NSObject]
        return Pet.findOne(query)
    }
    
}
